src/bmf.cpp

```diff
@@ -266,162 +261,159 @@ void CxadbmfPlayer::xadplayer_rewind(int subsong)
   if (bmf.version > BMF0_9B)
   {
     opl_write(0x01, 0x20);
-    
+
     /* 1.1 */
+    int i, j;
     if (bmf.version == BMF1_1)
-      for(i=0;i<9;i++)
-        for(j=0;j<13;j++)
-          opl_write(bmf_adlib_registers[13*i+j], bmf_default_instrument[j]);
+      for (i = 0; i < 9; i++)
+        for (j = 0; j < 13; j++)
+          opl_write(bmf_adlib_registers[13 * i + j], bmf_default_instrument[j]);
     /* 1.2 */
     else if (bmf.version == BMF1_2)
-      for(i=0x20; i<0x100; i++)
-        opl_write(i,0xFF); // very interesting, really!
+      for (i = 0x20; i < 0x100; i++)
+        opl_write(i, 0xFF); // very interesting, really!
   }
 
   /* ALL */
-
   opl_write(0x08, 0x00);
   opl_write(0xBD, 0xC0);
 }
 
 void CxadbmfPlayer::xadplayer_update()
 {
-  for(int i=0;i<9;i++)
-    if (bmf.channel[i].stream_position != 0xFFFF)
+  for (int i = 0; i < 9; i++)
+  {
+    unsigned short &pos = bmf.channel[i].stream_position;
+    if (pos == 0xFFFF)
+      continue;
+
     if (bmf.channel[i].delay)
+    {
       bmf.channel[i].delay--;
-	else
-	{
+      continue;
+    }
+
 #ifdef DEBUG
-   AdPlug_LogWrite("channel %02X:\n", i);
+    AdPlug_LogWrite("channel %02X:\n", i);
 #endif
-      bmf_event event;
 
-      // process so-called cross-events
-  	  while (true)
-	  {
-        memcpy(&event, &bmf.streams[i][bmf.channel[i].stream_position], sizeof(bmf_event));
+again:
+    const bmf_event &event = bmf.streams[i][pos];
 #ifdef DEBUG
-   AdPlug_LogWrite("%02X %02X %02X %02X %02X %02X\n",
-		   event.note,event.delay,event.volume,event.instrument,
-		   event.cmd,event.cmd_data);
+    AdPlug_LogWrite("%02X %02X %02X %02X %02X %02X\n",
+                    event.note, event.delay, event.volume,
+                    event.instrument, event.cmd, event.cmd_data);
```

Comment:
Personally I'm not a fan of excessive indenting without a very good reason, I'd be happy with normal indenting here.

Reply:
I think most LogWrites are formatted like this, but I really don't care.